src/mqt/yaqs/characterization/memory/backends/sequences/workers.py

```
@@ -0,0 +1,588 @@
+            msg = f"Choi feature rows must have length num_steps={num_steps}, got {rows.shape[0]}."
+            raise ValueError(msg)
+    else:
+        msg = f"Choi feature rows must be 1D or 2D, got ndim={rows.ndim}."
+        raise ValueError(msg)
+    return rows
+
+
+# ---------------------------------------------------------------------------
+# Sequence simulation core
+# ---------------------------------------------------------------------------
+def _simulate_seq_core(
```

Ruff (0.15.18)

[warning] Too many statements (76 > 50)

(PLR0915)